File: server/nestjs/src/base/usecase/product/product.usecase.ts
import { ProductUIVO, ProductUOVO } from "./vos";


export interface ProductUseCase {
    getProduct(id: number): Promise<ProductUOVO>;
    getAllProducts(): Promise<ProductUOVO[]>;
    createProduct(product: ProductUIVO): Promise<ProductUOVO>;
    updateProduct(id: number, product: ProductUIVO): Promise<ProductUOVO>;
    deleteProduct(id: number): Promise<void>;
    searchProductByNameAndPrice(name?: string, price?: number): Promise<ProductUOVO[]>;
  }

  export const ProductUseCaseToken = Symbol('ProductUseCase');

File: server/spring-kotlin/src/main/kotlin/com/usecase/product/usecase/ProductUseCase.kt
package com.usecase.product.usecase
import com.usecase.product.repository.vo.Product

interface ProductUseCase {
    fun createProduct(product: Product): Product
    fun updateProduct(product: Product): Product
    fun deleteProduct(id: Long)
    fun getProduct(id: Long): Product
    fun getProducts(): List<Product>
}